tt_cmap.c: Fix a bug in unicode encoding in CFF/OpenType fonts.
by Shunsaku Hirata  <shunsaku.hirata74@gmail.com>
upstream svn 34797
---
 texk/dvipdfm-x/ChangeLog |    4 ++++
 texk/dvipdfm-x/tt_cmap.c |    4 ++--
 2 files changed, 6 insertions(+), 2 deletions(-)

--- texlive-bin.orig/texk/dvipdfm-x/ChangeLog
+++ texlive-bin/texk/dvipdfm-x/ChangeLog
@@ -1,3 +1,7 @@
+2014-08-02  Shunsaku Hirata  <shunsaku.hirata74@gmail.com>
+
+	* tt_cmap.c: Fix a bug in unicode encoding in CFF/OpenType fonts.
+
 2014-07-24  Peter Breitenlohner  <peb@mppmu.mpg.de>
 
 	* pdffont.c (pdf_get_font): Drop unused variable.
--- texlive-bin.orig/texk/dvipdfm-x/tt_cmap.c
+++ texlive-bin/texk/dvipdfm-x/tt_cmap.c
@@ -798,7 +798,7 @@
 	       gid <= num_glyphs) {
 	  map[2*gid    ] = (cid >> 8) & 0xff;
 	  map[2*gid + 1] = cid & 0xff;
-	  gid++;
+	  gid++; cid++;
 	}
       }
     }
@@ -823,7 +823,7 @@
 		 gid <= num_glyphs) {
 	    map[gid] = (cid >> 8) & 0xff;
 	    map[gid] = cid & 0xff;
-	    gid++;
+	    gid++; cid++;
 	  }
 	}
 	
